shifthappens/tasks/imagenet_m/__init__.py

```diff
@@ -0,0 +1,180 @@
+"""Shift Happens task: ImageNet-M"""
+
+import dataclasses
+import json
+import os
+
+import numpy as np
+import torchvision.datasets.utils as tv_utils
+from torchvision import datasets as tv_datasets
+
+from shifthappens import benchmark as sh_benchmark
+from shifthappens.data import imagenet
+from shifthappens.models import base as sh_models
+from shifthappens.tasks.base import Task
+from shifthappens.tasks.metrics import Metric
+from shifthappens.tasks.task_result import TaskResult
+
+HUMAN_ACCURACY_JSON = "https://storage.googleapis.com/brain-car-datasets/imagenet-mistakes/human_accuracy_v3.0.0.json"
+CLASS_INFO_JSON = "https://raw.githubusercontent.com/modestyachts/ImageNetV2/master/data/metadata/class_info.json"
+IMAGENET_VALIDATION_LABELS = "https://raw.githubusercontent.com/tensorflow/datasets/master/tensorflow_datasets/image_classification/imagenet2012_validation_labels.txt"
+
+# If true, loads Greedy Soups model logits (pre-computed).
+# Eval should return 19 correct ImageNet-M examples.
+LOAD_GREEDYSOUPS_LOGITS = False
+GREEDYSOUPS_JSON = "https://storage.googleapis.com/brain-car-datasets/imagenet-mistakes/logits/greedysoups.npz"
+
+
+def load_npy_as_dict(path):
+    """Loads a numpy array of logits as a dictionary.
+
+    The returned dictionary maps ImageNet validation filenames
+    to a numpy array of logit values.
+    """
+    logits_array = np.load(open(path, "rb"))
+    ret = {}
+    for i in range(len(logits_array)):
+        filename = f"ILSVRC2012_val_{i+1:08}.JPEG".encode()
+        ret[filename] = logits_array[i]
+    return ret
+
+
+@sh_benchmark.register_task(
+    name="ImageNet-M", relative_data_folder="imagenet_m", standalone=True
+)
+@dataclasses.dataclass
+class ImageNetM(Task):
+    """Task definition for ImageNet-M, major mistakes eval subset.
+
+    This task evaluates a model on a small slice of the ImageNet
+    validation set.  The set has been additionally labeled with
+    multiple labels from the ImageNet label set by expert labelers.
+    The examples themselves are those where multiple high-performing
+    models in 2022 have yielded incorrect predictions where the
+    prediction was considered a "major" mistake.  
+    
+    The goal of this evaluation task is to measure a model's ability
+    to avoid making obvious mistakes; in theory, models should be
+    capable of scoring near 100% on this evaluation, in addition to
+    high performance on other image classification benchmarks.
+    See the citation file for more information about how the dataset 
+    was constructed.
+
+    This task uses a multi-label evaluation approach: if the top
+    prediction from the model is found in the set of correct
+    or unclear labels for each example, the prediction is considered
+    correct.
+
+    The metric computes the total correct / percentage of correctly
+    classified multi-label examples.
+    """
+    def setup(self):
```

Comment:
```suggestion
    """
    
    def setup(self):
```
Fix to pass black test.